OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -62,21 +61,48 @@ public IEnumerable<string> ProvidesPrerequisites
 			}
 		}
 
+		protected override void Created(Actor self)
+		{
+			techTree = self.Owner.PlayerActor.Trait<TechTree>();
+
+			Update(self.Owner, faction);
+
+			base.Created(self);
+		}
+
 		public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
-			if (info.ResetOnOwnerChange)
-				Update(newOwner, newOwner.Faction.InternalName);
+			techTree = newOwner.PlayerActor.Trait<TechTree>();
+
+			if (Info.ResetOnOwnerChange)
+				faction = newOwner.Faction.InternalName;
+
+			Update(newOwner, faction);
 		}
 
 		void Update(Player owner, string faction)
```

Comment:
This is now always called with `faction = this.faction`, so can you please remove this argument?

Reply:
done.